packages/styles/src/useStyles.ts

```diff
@@ -0,0 +1,116 @@
+import { useCallback, useContext, useMemo } from 'react';
+import { RendererContext, ThemeContext } from 'react-fela';
+import { G } from '@mobily/ts-belt';
+import { darken, lighten, transparentize } from 'polished';
+import { breakpointMediaQueries } from './breakpoints';
+import {
+    NativeStyle,
+    NativeStyleObject,
+    NativeStyleOrStylesParam,
+    NativeStyleUtils,
+    Style,
+    StyleOrStylesParam,
+    StyleUtils,
+} from './types';
+import { Theme, NativeTheme } from '@trezor/theme';
+import { multiply, getValueAndUnit, sum } from './utils';
+import { processStyles, processNativeStyles } from './processStyles';
+import { mediaQueries } from './mediaQueries';
+import { useDirection } from './useDirection';
+
+export const useStyles = () => {
+    const renderer = useContext(RendererContext);
+
+    if (!renderer) {
+        throw new Error('The `useStyles()` hook can only be used inside a `StylesProvider`.');
+    }
+
+    const theme = useContext(ThemeContext) as Theme;
```

Comment:
It will break typescript because typescript won't know state of that `isNative` flag.

But I will maybe split that useStyles into two hooks and separate `useNativeStyles`

Reply:
I ended with two hooks `useStyles` and `useNativeStyles` and it looks much better now.

Reply:
Yes, splitting is definitely good too! But I wonder, if the context provider had an `isNative` flag for example, would TS be able to notice that? It's not really that dynamic.

Reply:
TS could be aware it's there but there is no way how to make TS know if it's true/false which will depend on runtime evaluation. Anyway I don't see any benefits from introducing that type of flag.

Reply:
Yeah, I was just curious. 2 hooks makes more sense.